vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -133,9 +162,14 @@ def _set_categorical_selectors_options(self):
 
             self.selector.options = sorted(options)
 
+    def _set_filter_function(self):
+        default_filter_function = _filter_isin
+        filter_function = FILTER_FUNCTIONS.get(type(self.selector), default_filter_function)
+        return filter_function
+
     def _set_actions(self):
         if not self.selector.actions:
-            filter_function = _filter_between if isinstance(self.selector, RangeSlider) else _filter_isin
+            filter_function = self._set_filter_function()
```

Comment:
Check comment on trying out to insert datetime objects directly. If that works, then eventually this doesn't have to be changed?